src/main/node/config-manager.ts

```diff
@@ -1,144 +1,116 @@
-  if (isDirectory(path)) {
-    const configContent = {
-      name,
-      description,
-      collaborators,
-      packageManager,
-      tag,
-    };
-    const configContentJSON = JSON.stringify(configContent, null, 2);
-    fs.writeFileSync(`${path}/${configFileName}`, configContentJSON);
-    return true;
-  }
-  return false;
+  newCollaborators: Collaborator[]
+) {
+  validateNonEmpty(newCollaborators);
+  const configurationToUpdate: Configuration = getConfigFile(path);
+  updateConfigField(configurationToUpdate, path, () => {
+    configurationToUpdate.collaborators = newCollaborators;
+  });
 }
 
-export function deleteConfigFile(path: string): boolean {
-  if (isDirectory(path) && isPublication(path)) {
-    try {
-      if (fs.existsSync(path)) {
-        fs.unlinkSync(`${path}/${configFileName}`);
-        return true;
-      }
-    } catch (err) {
-      console.error(err);
-      return false;
-    }
-  }
-  return false;
+export function updatePackageManager(path: string, newPackageManager: string) {
+  validateNonEmptyNorNull(newPackageManager);
+  const configurationToUpdate: Configuration = getConfigFile(path);
+  updateConfigField(configurationToUpdate, path, () => {
+    configurationToUpdate.packageManager = newPackageManager;
+  });
 }
 
-// TODO: Unsure how much sense it makes to have all separate `modify` functions, could probably abstract that
-export function modifyName(path: string, newName: string): boolean {
-  if (isDirectory(path) && isPublication(path)) {
-    const configContentJSON = fs.readFileSync(`${path}/${configFileName}`);
-    const configContent = JSON.parse(configContentJSON);
-    configContent.name = newName;
-    const newConfigContentJSON = JSON.stringify(configContent);
-    deleteConfigFile(path);
-    fs.writeFileSync(`${path}/${configFileName}`, newConfigContentJSON);
-    return true;
-  }
-  return false;
+export function updateTag(path: string, tag: string) {
+  const configurationToUpdate: Configuration = getConfigFile(path);
+  updateConfigField(configurationToUpdate, path, () => {
+    configurationToUpdate.tag = tag;
+  });
 }
 
-export function modifyDescription(
-  path: string,
-  newDescription: string
-): boolean {
-  if (isDirectory(path) && isPublication(path)) {
-    const configContentJSON = fs.readFileSync(`${path}/${configFileName}`);
-    const configContent = JSON.parse(configContentJSON);
-    configContent.description = newDescription;
-    const newConfigContentJSON = JSON.stringify(configContent);
-    deleteConfigFile(path);
-    fs.writeFileSync(`${path}/${configFileName}`, newConfigContentJSON);
-    return true;
-  }
-  return false;
+export function getConfigFile(path: string): Configuration {
+  validatePath(
+    !isDirectory(path) || !isPublication(path),
+    'Config file does not exist. Check the provided path'
+  );
+  const configContentJSON = fs.readFileSync(`${path}/${configFileName}`);
+  return JSON.parse(configContentJSON) as Configuration;
```

Comment:
In case this line gives you errors after changing import type (since the `readFileSync` return `Buffer` and `parse` expects a `string`) you can just do `JSON.parse(configContentJSON.toString())`.